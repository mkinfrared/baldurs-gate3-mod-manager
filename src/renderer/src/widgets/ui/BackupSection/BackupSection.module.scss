.BackupSection {
  .buttonGroup {
    display: flex;
    gap: var(--spacing-5);
    margin-block: var(--spacing-6) var(--spacing-12);
  }
}
