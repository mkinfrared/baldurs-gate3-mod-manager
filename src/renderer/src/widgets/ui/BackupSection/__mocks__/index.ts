export { BackupSectionMock as BackupSection } from "./BackupSection.mock";
