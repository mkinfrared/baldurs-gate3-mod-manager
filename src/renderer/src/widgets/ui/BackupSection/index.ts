export { BackupSection } from "./BackupSection";

export * from "./BackupSection.type";
