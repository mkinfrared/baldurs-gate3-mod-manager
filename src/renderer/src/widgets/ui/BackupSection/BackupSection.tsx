import {
  BackupList,
  RestoreDefaultSettings,
  SaveSettingsButton,
} from "@renderer/features/backup";
import { SuspenseError } from "@renderer/shared/components";
import { classNames } from "@renderer/shared/lib/helpers";

import css from "./BackupSection.module.scss";
import { BackupSectionProps } from "./BackupSection.type";

const BackupSection = ({ className, game }: BackupSectionProps) => (
  <div
    className={classNames(css.BackupSection, className)}
    data-testid="BackupSection"
  >
    <div className={css.buttonGroup}>
      <SaveSettingsButton game={game} />
      <RestoreDefaultSettings game={game} />
    </div>
    <SuspenseError>
      <BackupList game={game} />
    </SuspenseError>
  </div>
);

export { BackupSection };
