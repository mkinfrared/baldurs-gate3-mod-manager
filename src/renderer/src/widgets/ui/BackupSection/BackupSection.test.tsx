import { render } from "@testing-library/react";

import { BackupSection } from "./BackupSection";

vi.mock("@renderer/features/backup");

describe("<BackupSection />", () => {
  const game = "BG3";
  const Component = <BackupSection game={game} />;

  it("should be defined", () => {
    expect(BackupSection).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("BackupSection");

    expect(element).toBeDefined();
  });
});
