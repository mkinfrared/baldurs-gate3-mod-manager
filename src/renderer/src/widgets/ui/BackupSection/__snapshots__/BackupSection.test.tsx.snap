// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<BackupSection /> > should match the snapshot 1`] = `
<div>
  <div
    class="BackupSection"
    data-testid="BackupSection"
  >
    <div
      class="buttonGroup"
    />
  </div>
</div>
`;
